internal/controller/clientpool/clientpool.go

```diff
@@ -25,17 +29,23 @@ type ClientPoolKey struct {
 	Namespace string
 }
 
+type ClientInfo struct {
+	Client  sdkclient.Client
+	TLS     *tls.Config // Storing the TLS config associated with the client to check certificate expiration. If the certificate is expired, a new client will be created.
+	PEMCert []byte      // Store the original PEM-encoded certificate for expiration checks
```

Comment:
Would it make sense to only store the expiration time instead of the whole cert?